make build fails during linking liblumen_llvm in lumen on Linux

When using the [Dockerfile](https://github.com/lumen/lumen/blob/3c32ab5be6ad0de4ba65ebbf5b5ef08c575813bf/.github/workflows/Dockerfile) to try to `make build` for `x86_64-unknown-linux-gnu-compiler` workflow, the build fails when during compiling `lumen` because the linker can't find symbols for `liblumen_llvm`:

<details>

 "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64" "-L" "/usr/local/rustup/toolchains/nightly-2020-05-13-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "/code/target/debug/deps/lumen-9f105fb1949cc411.1abwt9d1bbbit8vt.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.1mm2ai2o28ss4j1f.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.1mm8pk8z0getakmr.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.2f58ka6l9pso1bv2.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.2op8328kqpflkeij.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.2r1q134z9ij9bnr8.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.2use12i4s6bwrywj.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.2w4hq3i050e1z202.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.2z0aroab2zk74j4g.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.314gv311pngszzhy.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.37e8j973oe7dmrt3.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.37fr66df17nf1rxg.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.3a7rd7y5pdpbzreh.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.41lcb4zylj6phxrw.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.42e84xjkjv6xjqvz.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.45ijwo9lx004y7qw.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.45pja9qlqrgomdqy.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.479lnp2dlekda8ug.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.4kib6bt5tewukvyq.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.4mq2nu4t9k2ev0wl.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.4na1xcyt5n0b0u5c.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.4y5z3lf0khclebec.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.523baj3p9ayodrx5.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.7u8a2trco80jrnh.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.o4naln9ce8eqpyu.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.pqbgpvc90hen67s.rcgu.o" "/code/target/debug/deps/lumen-9f105fb1949cc411.u1xx1ycxwxb7j3s.rcgu.o" "-o" "/code/target/debug/deps/lumen-9f105fb1949cc411" "/code/target/debug/deps/lumen-9f105fb1949cc411.4qu7b8977dgi60o2.rcgu.o" "-Wl,--gc-sections" "-pie" "-Wl,-zrelro" "-Wl,-znow" "-nodefaultlibs" "-L" "/code/target/debug/deps" "-L" "/code/target/debug/build/backtrace-sys-b95fcfb4fc6a7e5b/out" "-L" "/code/target/debug/build/liblumen_codegen-1e31d227776baf0c/out/lib" "-L" "/code/target/debug/build/liblumen_llvm-c0e6a65a948cf5ab/out" "-L" "/opt/llvm/lib" "-L" "/code/target/debug/build/liblumen_mlir-35d91c088acb37a4/out" "-L" "/opt/llvm/lib" "-L" "/usr/local/rustup/toolchains/nightly-2020-05-13-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/code/target/debug/deps/libenv_logger-ba787a8604587f47.rlib" "/code/target/debug/deps/libhumantime-1c6a436c85debe2d.rlib" "/code/target/debug/deps/libquick_error-d17ee97fa9f336e0.rlib" "/code/target/debug/deps/libhuman_panic-93c7e3479381aa12.rlib" "/code/target/debug/deps/libuuid-42ecf84a6ae45b0f.rlib" "/code/target/debug/deps/libtoml-b18a1c677f226540.rlib" "/code/target/debug/deps/libtempdir-e7fc9983e37ca695.rlib" "/code/target/debug/deps/librand-d6b5f058bb2377bf.rlib" "/code/target/debug/deps/libos_type-6b4368c235a62bdd.rlib" "/code/target/debug/deps/libtermcolor-a7daedd6fd89a4d3.rlib" "/code/target/debug/deps/libfailure-d6610b30b30a2020.rlib" "/code/target/debug/deps/libliblumen_compiler-0ca1422daf278c44.rlib" "/code/target/debug/deps/libasync_task-bf9b82bceed116bf.rlib" "/code/target/debug/deps/libfutures-357fe188829b114c.rlib" "/code/target/debug/deps/libfutures_executor-d050407232ffc49d.rlib" "/code/target/debug/deps/libfutures_util-873ccbeae4e285ea.rlib" "/code/target/debug/deps/libproc_macro_nested-f004ccefb0366f47.rlib" "/code/target/debug/deps/libfutures_io-fd2902bb42d397cd.rlib" "/code/target/debug/deps/libslab-881f6e1e2517fedf.rlib" "/code/target/debug/deps/libfutures_channel-8dcbc7f5dd7aea8c.rlib" "/code/target/debug/deps/libfutures_sink-73af31634f99260f.rlib" "/code/target/debug/deps/libfutures_task-4919351dc8e7a893.rlib" "/code/target/debug/deps/libpin_utils-84e22046e2801905.rlib" "/code/target/debug/deps/libfutures_core-85d17b193f942900.rlib" "/code/target/debug/deps/libwalkdir-526e7d274633ba41.rlib" "/code/target/debug/deps/libsame_file-8212f6796759b052.rlib" "/code/target/debug/deps/liblibeir_passes-3aa300afb9327d48.rlib" "/code/target/debug/deps/liblibeir_util_pattern_compiler-31919ccf1704c75f.rlib" "/code/target/debug/deps/liblibeir_frontend-daa02c9723a4528a.rlib" "/code/target/debug/deps/libsalsa-cf9b3d979484185f.rlib" "/code/target/debug/deps/libindexmap-b5363901e8b9ea4a.rlib" "/code/target/debug/deps/libcrossbeam-a712c5ae218bf4aa.rlib" "/code/target/debug/deps/libcrossbeam_queue-28327da2ab24d829.rlib" "/code/target/debug/deps/libcrossbeam_channel-70f0330daadeacd2.rlib" "/code/target/debug/deps/libcrossbeam_deque-093024c769158c96.rlib" "/code/target/debug/deps/libcrossbeam_epoch-9a591bbcd5f337bc.rlib" "/code/target/debug/deps/libmemoffset-8b78cd7f5ab1fbbd.rlib" "/code/target/debug/deps/libcrossbeam_utils-23e351823d7f0270.rlib" "/code/target/debug/deps/libparking_lot-5fbb3e3e01eaa15b.rlib" "/code/target/debug/deps/libparking_lot_core-20afc89f8433200e.rlib" "/code/target/debug/deps/libsmallvec-5cf5ec7023830aa2.rlib" "/code/target/debug/deps/libliblumen_codegen-f441b746864ef570.rlib" "/code/target/debug/deps/libtempfile-f06507b44edb8b91.rlib" "/code/target/debug/deps/librand-e59d438b0e5c94c7.rlib" "/code/target/debug/deps/librand_pcg-68c0710915fa6da8.rlib" "/code/target/debug/deps/librand_chacha-062a00bc2b428611.rlib" "/code/target/debug/deps/libc2_chacha-ccf5b3bb8dcd2c69.rlib" "/code/target/debug/deps/libppv_lite86-46965f3209b0f83f.rlib" "/code/target/debug/deps/librand_core-f77b4a18cc7641fb.rlib" "/code/target/debug/deps/libgetrandom-5072462c8f8e22c6.rlib" "/code/target/debug/deps/libremove_dir_all-7819f6625b9533ee.rlib" "/code/target/debug/deps/libcc-6aa194558d4047b4.rlib" "/code/target/debug/deps/libliblumen_alloc-ad8151cc65bc710e.rlib" "/code/target/debug/deps/libheapless-0831910637fa726b.rlib" "/code/target/debug/deps/libas_slice-c497312fc3e310b0.rlib" "/code/target/debug/deps/libstable_deref_trait-42f6d0b42bd36740.rlib" "/code/target/debug/deps/libgeneric_array-8a0d809d49cfedf4.rlib" "/code/target/debug/deps/libhash32-f8c6e94fba01ea1c.rlib" "/code/target/debug/deps/libgeneric_array-bfbac7176276bc4c.rlib" "/code/target/debug/deps/libtypenum-c3aa31e10b848e73.rlib" "/code/target/debug/deps/libliblumen_arena-7c551a57c6981f99.rlib" "/code/target/debug/deps/libonce_cell-0a56a49a1df330d7.rlib" "/code/target/debug/deps/libintrusive_collections-ed2dba98817fdd25.rlib" "/code/target/debug/deps/libstatic_assertions-91cb8b3854d261ca.rlib" "/code/target/debug/deps/libnum_bigint-121466e82de9a69e.rlib" "/code/target/debug/deps/liblibeir_lowerutils-7a01bc33e00c8e05.rlib" "/code/target/debug/deps/libliblumen_core-57cc40a36789b846.rlib" "/code/target/debug/deps/libparking_lot-61430dbe3a2165ab.rlib" "/code/target/debug/deps/libparking_lot_core-3516b157d4f44f8c.rlib" "/code/target/debug/deps/liblock_api-de5c966646724987.rlib" "/code/target/debug/deps/libliblumen_mlir-76d541cf29a140c9.rlib" "/code/target/debug/deps/libliblumen_llvm-413fa2c7687a2935.rlib" "/code/target/debug/deps/libliblumen_profiling-ef8e65ff2dda2c74.rlib" "/code/target/debug/deps/libmemmap-77940a8065666065.rlib" "/code/target/debug/deps/libparking_lot-c2d6b138634af539.rlib" "/code/target/debug/deps/libparking_lot_core-c8286e2c9cfaf3f2.rlib" "/code/target/debug/deps/libsmallvec-9ad6cd8a2114f03e.rlib" "/code/target/debug/deps/libmaybe_uninit-1f5b425bcc9783e4.rlib" "/code/target/debug/deps/liblock_api-b61939d454a30794.rlib" "/code/target/debug/deps/libscopeguard-fc4a68f946596087.rlib" "/code/target/debug/deps/libfxhash-b8521a9bb6dc1998.rlib" "/code/target/debug/deps/libliblumen_session-d4b0c550750fec93.rlib" "/code/target/debug/deps/liblibeir_syntax_erl-cfbc86d3dd785af1.rlib" "/code/target/debug/deps/libglob-98b9183b1496b66c.rlib" "/code/target/debug/deps/liblibeir_util_parse_listing-367b8e62c4b3eb8e.rlib" "/code/target/debug/deps/liblibeir_ir-2426ebe51d70440e.rlib" "/code/target/debug/deps/libregex-2371beb6321cbffd.rlib" "/code/target/debug/deps/libthread_local-6970bd75f2274a2e.rlib" "/code/target/debug/deps/libregex_syntax-c678d83133223e59.rlib" "/code/target/debug/deps/libaho_corasick-c2b69f44b9c3e22a.rlib" "/code/target/debug/deps/libmemchr-e0167a688bd7052f.rlib" "/code/target/debug/deps/libitertools-b70cd26b9184c4be.rlib" "/code/target/debug/deps/libeither-699fb31a0c727ddd.rlib" "/code/target/debug/deps/liblibeir_util_number-ca9d902cf9d3380b.rlib" "/code/target/debug/deps/libnum_bigint-cc27a32fa64867ca.rlib" "/code/target/debug/deps/libnum_integer-cc2ddbc866375112.rlib" "/code/target/debug/deps/libnum_traits-d2684d67ce3a5a5f.rlib" "/code/target/debug/deps/liblalrpop_util-4876fc5214749a64.rlib" "/code/target/debug/deps/liblibeir_util_parse-2138aa9b17a5febb.rlib" "/code/target/debug/deps/libscoped_cell-44fb434cac8772c2.rlib" "/code/target/debug/deps/libfnv-a77cfc28d518fe21.rlib" "/code/target/debug/deps/libbumpalo-99745e28b26814b9.rlib" "/code/target/debug/deps/libpetgraph-f68307c20062d061.rlib" "/code/target/debug/deps/libordermap-53c69527569731ff.rlib" "/code/target/debug/deps/libfixedbitset-6d18236bd70a0b1f.rlib" "/code/target/debug/deps/libsnafu-7be97f7e9b3089ca.rlib" "/code/target/debug/deps/libdoc_comment-2e4b49608de14791.rlib" "/code/target/debug/deps/libbacktrace-7ab6efb34bbcefc8.rlib" "/code/target/debug/deps/libbacktrace_sys-7a75f0358dc46569.rlib" "/code/target/debug/deps/librustc_demangle-0324586c470a773b.rlib" "/code/target/debug/deps/libpretty-e5dcc8e22c70f58c.rlib" "/code/target/debug/deps/libtyped_arena-5e0d46ab4c382db7.rlib" "/code/target/debug/deps/libstack_dst-cea4b12fb686ce02.rlib" "/code/target/debug/deps/libmeta_table-5853f1fa15559051.rlib" "/code/target/debug/deps/liblibeir_util_dot_graph-dca2f64ce433e257.rlib" "/code/target/debug/deps/libserde-e1e05803a369bf9d.rlib" "/code/target/debug/deps/liblibeir_util_datastructures-e5d4720bad4d5dac.rlib" "/code/target/debug/deps/libhashbrown-f747e83fbf7cc1cd.rlib" "/code/target/debug/deps/libcranelift_entity-17595da3f8df00a0.rlib" "/code/target/debug/deps/libcranelift_bforest-e3aa058f16dfc4ff.rlib" "/code/target/debug/deps/libcranelift_entity-db2a2f022161ca90.rlib" "/code/target/debug/deps/liblibeir_intern-1153266137d7b004.rlib" "/code/target/debug/deps/librustc_hash-27aaba25741e2922.rlib" "/code/target/debug/deps/libbyteorder-e2066de2cc8ed254.rlib" "/code/target/debug/deps/liblazy_static-74222b977504fcd9.rlib" "/code/target/debug/deps/libliblumen_target-6d05c3f3bc274fae.rlib" "/code/target/debug/deps/libliblumen_term-9aeac7c033fcdff2.rlib" "/code/target/debug/deps/libliblumen_util-fb68e4312bd25aa5.rlib" "/code/target/debug/deps/libglob-b32fb543d938540e.rlib" "/code/target/debug/deps/libthiserror-3bc4bb3d02fcd6ca.rlib" "/code/target/debug/deps/liblibeir_diagnostics-5cee099423ace1b4.rlib" "/code/target/debug/deps/libcodespan-2f904ecc1f7274bb.rlib" "/code/target/debug/deps/libcodespan_reporting-6f7f86d447121816.rlib" "/code/target/debug/deps/libtermcolor-cc79fc8789b69011.rlib" "/code/target/debug/deps/libdashmap-6541ebed6c1d8856.rlib" "/code/target/debug/deps/libnum_cpus-4757bc2301425b6c.rlib" "/code/target/debug/deps/libhashbrown-d8d9eb1d31f275ef.rlib" "/code/target/debug/deps/libahash-526ec29835f5313f.rlib" "/code/target/debug/deps/libconst_random-0818ac27c34e7448.rlib" "/code/target/debug/deps/libclap-9e91577e5993e430.rlib" "/code/target/debug/deps/libvec_map-a6d95081468d0c5b.rlib" "/code/target/debug/deps/libtextwrap-1cc32fc367d9b6dd.rlib" "/code/target/debug/deps/libunicode_width-7568678480dd7051.rlib" "/code/target/debug/deps/libstrsim-d5d4db4e2d4c408f.rlib" "/code/target/debug/deps/libbitflags-6c14cce4865eaed0.rlib" "/code/target/debug/deps/libatty-e0ef74ec7b78c092.rlib" "/code/target/debug/deps/liblibc-4b8cef420788eda8.rlib" "/code/target/debug/deps/libansi_term-7a570c1260eb316a.rlib" "/code/target/debug/deps/libanyhow-aafd9c27011304cc.rlib" "/code/target/debug/deps/liblog-2ff13020d2e94f5d.rlib" "/code/target/debug/deps/libcfg_if-f1737ba34a598554.rlib" "-Wl,--start-group" "/usr/local/rustup/toolchains/nightly-2020-05-13-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-0126adaa107032a6.rlib" "/usr/local/rustup/toolchains/nightly-2020-05-13-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-57961a3692c39401.rlib" "/usr/local/rustup/toolchains/nightly-2020-05-13-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libhashbrown-9fa9e50746f44a4c.rlib" "/usr/local/rustup/toolchains/nightly-2020-05-13-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_alloc-251012178f8196c3.rlib" "/usr/local/rustup/toolchains/nightly-2020-05-13-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libbacktrace-cafd612ee1b4bf88.rlib" "/usr/local/rustup/toolchains/nightly-2020-05-13-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libbacktrace_sys-140b53dc81f0d53c.rlib" "/usr/local/rustup/toolchains/nightly-2020-05-13-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_demangle-cbaa12f3d93cc54c.rlib" "/usr/local/rustup/toolchains/nightly-2020-05-13-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-a5bcc2d112e5a391.rlib" "/usr/local/rustup/toolchains/nightly-2020-05-13-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcfg_if-5ada18586d2652e2.rlib" "/usr/local/rustup/toolchains/nightly-2020-05-13-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-ac41f78688451395.rlib" "/usr/local/rustup/toolchains/nightly-2020-05-13-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-b1966f66603f91b6.rlib" "/usr/local/rustup/toolchains/nightly-2020-05-13-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_core-428d15b56101bdc7.rlib" "/usr/local/rustup/toolchains/nightly-2020-05-13-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-941f7f6bc622e313.rlib" "-Wl,--end-group" "/usr/local/rustup/toolchains/nightly-2020-05-13-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-e915d31ab7edbbd4.rlib" "-Wl,-Bdynamic" "-llumen_compiler_Dialect_EIR_IR_IR" "-llumen_compiler_Dialect_EIR_Conversion_EIRToLLVM_EIRToLLVM" "-llumen_compiler_Dialect_EIR_Transforms_Transforms" "-llumen_compiler_Target_Target" "-llumen_compiler_Translation_Translation" "-lMLIRAnalysis" "-lMLIRAffine" "-lMLIRCallInterfaces" "-lMLIRControlFlowInterfaces" "-lMLIRDerivedAttributeOpInterface" "-lMLIRDialect" "-lMLIREDSC" "-lMLIREDSCInterface" "-lMLIRExecutionEngine" "-lMLIRIR" "-lMLIRInferTypeOpInterface" "-lMLIRLLVMIR" "-lMLIRLLVMIRTransforms" "-lMLIRLoopAnalysis" "-lMLIRLoopOps" "-lMLIRLoopLikeInterface" "-lMLIROpenMP" "-lMLIRParser" "-lMLIRPass" "-lMLIRSideEffects" "-lMLIRStandardOps" "-lMLIRStandardToLLVM" "-lMLIRSupport" "-lMLIRTargetLLVMIR" "-lMLIRTargetLLVMIRModuleTranslation" "-lMLIRTransformUtils" "-lMLIRTransforms" "-lMLIRTranslation" "-lrt" "-ldl" "-lpthread" "-lm" "-lxml2" "-lc++" "-lutil" "-ldl" "-lutil" "-ldl" "-lrt" "-lpthread" "-lgcc_s" "-lc" "-lm" "-lrt" "-lpthread" "-lutil" "-ldl" "-lutil" "-Wl,-rpath,/usr/local/rustup/toolchains/nightly-2020-05-13-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib"
  = note: /usr/bin/ld: /code/target/debug/deps/libliblumen_llvm-413fa2c7687a2935.rlib(X86TargetMachine.cpp.o): in function `LLVMInitializeX86Target':
          X86TargetMachine.cpp:(.text.LLVMInitializeX86Target+0x8c): undefined reference to `std::__once_callable'
...
<details>

--- a/.github/workflows/Dockerfile
+++ b/.github/workflows/Dockerfile
@@ -5,13 +5,13 @@ RUN cd /tmp \
   && tar -xvzf clang+llvm-10.0.0-x86_64-linux-gnu.tar.gz \
   && rm -f clang+llvm-10.0.0-x86_64-linux-gnu.tar.gz \
   && mv clang+llvm-10.0.0-x86_64-linux-gnu /opt/llvm \
-  && ln -s clang /opt/llvm/bin/cc \
-  && ln -s lld /opt/llvm/bin/ld
+  && update-alternatives --install /usr/bin/cc cc /opt/llvm/bin/clang 90 \
+  && update-alternatives --install /usr/bin/c++ c++ /opt/llvm/bin/clang++ 90 \
+  && update-alternatives --install /usr/bin/cpp cpp /opt/llvm/bin/clang++ 90 \
+  && update-alternatives --install /usr/bin/ld ld /opt/llvm/bin/lld 90
 
 ENV PATH="/opt/llvm/bin:${PATH}"
 ENV LLVM_PREFIX="/opt/llvm"
-ENV CC=/opt/llvm/bin/clang
-ENV CXX=/opt/llvm/bin/clang++
 
 RUN apt-get update \
   && apt-get install -y cmake jq libc++-dev libc++abi-dev libunwind-dev ninja-build rsync

--- a/.github/workflows/wasm32-unknown-unknown-linux.yml
+++ b/.github/workflows/wasm32-unknown-unknown-linux.yml
@@ -6,7 +6,7 @@ jobs:
   wasm:
     runs-on: ubuntu-latest
     container:
-      image: kronicdeth/lumen-development@sha256:adee04bab3a607633d8550235b261a6412524be8f8ad0b0189d97ff9e2100ccd
+      image: kronicdeth/lumen-development@sha256:8625315ed528ac92ebbd31023a333755a9325d51f042f02f393877d0356e20ff
 
     steps:
       - uses: actions/checkout@v2

--- a/.github/workflows/wasm32-unknown-unknown-linux.yml
+++ b/.github/workflows/wasm32-unknown-unknown-linux.yml
@@ -6,7 +6,7 @@ jobs:
   wasm:
     runs-on: ubuntu-latest
     container:
-      image: kronicdeth/lumen-development@sha256:8625315ed528ac92ebbd31023a333755a9325d51f042f02f393877d0356e20ff
+      image: kronicdeth/lumen-development@sha256:985b21718f2d290db7efb585046161ccf5116d046e90cae922985a02e3a3d8bf
 
     steps:
       - uses: actions/checkout@v2

--- a/.github/workflows/x86_64-unknown-linux-gnu-compiler.yml
+++ b/.github/workflows/x86_64-unknown-linux-gnu-compiler.yml
@@ -1,0 +1,32 @@
+name: x86_64-unknown-linux-gnu compiler
+
+on: push
+
+jobs:
+  compiler:
+    runs-on: ubuntu-latest
+    container: kronicdeth/lumen-development@sha256:985b21718f2d290db7efb585046161ccf5116d046e90cae922985a02e3a3d8bf
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache cargo registry
+        uses: actions/cache@v1
+        with:
+          path: ~/.cargo/registry
+          key: ${{ runner.os }}-formatted-cargo-registry-${{ hashFiles('**/Cargo.lock') }}
+      - name: Cache cargo index
+        uses: actions/cache@v1
+        with:
+          path: ~/.cargo/git
+          key: ${{ runner.os }}-formatted-cargo-index-${{ hashFiles('**/Cargo.lock') }}
+      - name: Cache cargo build
+        uses: actions/cache@v1
+        with:
+          path: target
+          key: ${{ runner.os }}-formatted-cargo-build-target-${{ hashFiles('**/Cargo.lock') }}
+      - name: Make Build
+        env:
+          RUST_BACKTRACE: full
+        run: make build
+      - name: Lumen Test
+        run: cargo test --package lumen

--- a/.github/workflows/x86_64-unknown-linux-gnu-libraries.yml
+++ b/.github/workflows/x86_64-unknown-linux-gnu-libraries.yml
@@ -5,7 +5,7 @@ on: push
 jobs:
   formatted:
     runs-on: ubuntu-latest
-    container: kronicdeth/lumen-development@sha256:8625315ed528ac92ebbd31023a333755a9325d51f042f02f393877d0356e20ff
+    container: kronicdeth/lumen-development@sha256:985b21718f2d290db7efb585046161ccf5116d046e90cae922985a02e3a3d8bf
 
     steps:
       - uses: actions/checkout@v2
@@ -29,7 +29,7 @@ jobs:
 
   libraries:
     runs-on: ubuntu-latest
-    container: kronicdeth/lumen-development@sha256:8625315ed528ac92ebbd31023a333755a9325d51f042f02f393877d0356e20ff
+    container: kronicdeth/lumen-development@sha256:985b21718f2d290db7efb585046161ccf5116d046e90cae922985a02e3a3d8bf
 
     steps:
       - uses: actions/checkout@v2

--- a/.github/workflows/x86_64-unknown-linux-gnu-libraries.yml
+++ b/.github/workflows/x86_64-unknown-linux-gnu-libraries.yml
@@ -5,7 +5,7 @@ on: push
 jobs:
   formatted:
     runs-on: ubuntu-latest
-    container: kronicdeth/lumen-development@sha256:adee04bab3a607633d8550235b261a6412524be8f8ad0b0189d97ff9e2100ccd
+    container: kronicdeth/lumen-development@sha256:8625315ed528ac92ebbd31023a333755a9325d51f042f02f393877d0356e20ff
 
     steps:
       - uses: actions/checkout@v2
@@ -29,7 +29,7 @@ jobs:
 
   libraries:
     runs-on: ubuntu-latest
-    container: kronicdeth/lumen-development@sha256:adee04bab3a607633d8550235b261a6412524be8f8ad0b0189d97ff9e2100ccd
+    container: kronicdeth/lumen-development@sha256:8625315ed528ac92ebbd31023a333755a9325d51f042f02f393877d0356e20ff
 
     steps:
       - uses: actions/checkout@v2

--- a/.github/workflows/x86_64-unknown-linux-gnu-runtime_full.yml
+++ b/.github/workflows/x86_64-unknown-linux-gnu-runtime_full.yml
@@ -5,7 +5,7 @@ on: push
 jobs:
   runtime:
     runs-on: ubuntu-latest
-    container: kronicdeth/lumen-development@sha256:8625315ed528ac92ebbd31023a333755a9325d51f042f02f393877d0356e20ff
+    container: kronicdeth/lumen-development@sha256:985b21718f2d290db7efb585046161ccf5116d046e90cae922985a02e3a3d8bf
 
     steps:
       - uses: actions/checkout@v2

--- a/.github/workflows/x86_64-unknown-linux-gnu-runtime_full.yml
+++ b/.github/workflows/x86_64-unknown-linux-gnu-runtime_full.yml
@@ -5,7 +5,7 @@ on: push
 jobs:
   runtime:
     runs-on: ubuntu-latest
-    container: kronicdeth/lumen-development@sha256:adee04bab3a607633d8550235b261a6412524be8f8ad0b0189d97ff9e2100ccd
+    container: kronicdeth/lumen-development@sha256:8625315ed528ac92ebbd31023a333755a9325d51f042f02f393877d0356e20ff
 
     steps:
       - uses: actions/checkout@v2

--- a/compiler/llvm/build.rs
+++ b/compiler/llvm/build.rs
@@ -239,7 +239,6 @@ fn main() {
     println!("cargo:ldflags={}", shared_ldflags.as_slice().join(";"));
 
     let llvm_static_stdcpp = env::var_os("LLVM_STATIC_STDCPP");
-    let llvm_use_libcxx = env::var_os("LLVM_USE_LIBCXX");
 
     let stdcppname = if target.contains("openbsd") {
         if target.contains("sparc64") {
@@ -254,8 +253,6 @@ fn main() {
     } else if target.contains("netbsd") && llvm_static_stdcpp.is_some() {
         // NetBSD uses a separate library when relocation is required
         "stdc++_pic"
-    } else if llvm_use_libcxx.is_some() {
-        "c++"
     } else {
         "stdc++"
     };

